src/Command/CleanUpCommand.php

```diff
@@ -2,63 +2,48 @@
 
 namespace Diag\Command;
 
-use Diag\Config;
 use Diag\Storage\CanCleanUp;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Output\OutputInterface;
 
-class CleanUpCommand extends Command
+class CleanUpCommand extends ContainerAwareCommand
 {
     protected function configure()
     {
         $this
             ->setName('maintenance:clean-up')
-            ->addOption(
-                'storage',
-                's',
-                InputArgument::OPTIONAL,
-                'storage engine to use',
-                getenv('DIAG_DEFAULT_STORAGE')
-            )
             ->addOption(
                 'now',
                 null,
                 InputArgument::OPTIONAL,
                 'time string to use as now',
-                null
-            )
-
-        ;
+                'now'
+            );
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $config = new Config();
-
-        $storageClassName = '\\Diag\\Storage\\' . $input->getOption('storage');
-        if (
-            !preg_match('/^[a-z]+$/i', $input->getOption('storage')) ||
-            !$config->hasStorage($input->getOption('storage')) ||
-            !class_exists($storageClassName)
-        ) {
-            throw new \RuntimeException('storage not supported');
-        }
+        $storageClassName = $this->container->getParameter('diag.storage');
+        $storage = $this->container->get($storageClassName);
 
-        $storage = new $storageClassName($config);
         if (!($storage instanceof CanCleanUp)) {
             $output->writeln('storage does not support clean up');
-            return 1;
+            return 10;
```

Comment:
Wouldn't it be better to throw exceptions instead? Instead of returning error codes.